src/NLog/Targets/FileTarget.cs

```diff
@@ -1096,8 +1100,13 @@ protected virtual string GetFormattedMessage(LogEventInfo logEvent)
         /// <returns>Array of bytes that are ready to be written.</returns>
         protected virtual byte[] GetBytesToWrite(LogEventInfo logEvent)
         {
-            string renderedText = this.GetFormattedMessage(logEvent) + this.NewLineChars;
-            return this.TransformBytes(this.Encoding.GetBytes(renderedText));
+            string text = this.GetFormattedMessage(logEvent);
+            int textBytesCount = this.Encoding.GetByteCount(text);
+            int newLineBytesCount = this.Encoding.GetByteCount(this.NewLineChars);
+            byte[] bytes = new byte[textBytesCount + newLineBytesCount];
+            this.Encoding.GetBytes(text, 0, text.Length, bytes, 0);
+            this.Encoding.GetBytes(this.NewLineChars, 0, this.NewLineChars.Length, bytes, textBytesCount);
```

Comment:
@304NotModified 

I wrote a version where 'line ending bytes' are computed only on setting Encoding/LineEnding property of FileTarget (as i understand user can change this properties at runtime in some way?). 
My tests on small and medium message sizes show same throughput as in this implementation. 
So i think we shouldn't add more complexity to codebase, because there isn't any significant benefits.